clients/admin-ui/src/features/dataset/EditDrawer.tsx

```diff
@@ -46,30 +52,42 @@ const EditDrawer = ({
     <>
       <Drawer placement="right" isOpen={isOpen} onClose={onClose} size="lg">
         <DrawerOverlay />
-        <DrawerContent data-testid="edit-drawer-content">
-          <Box py={2}>
-            <Box display="flex" justifyContent="flex-end" mr={2}>
-              <Button variant="ghost" onClick={onClose}>
-                <CloseSolidIcon />
-              </Button>
-            </Box>
-            <DrawerHeader py={2} display="flex" alignItems="center">
-              <Text mr="2">{header}</Text>
-              <IconButton
-                aria-label="delete"
-                icon={<TrashCanSolidIcon />}
-                size="xs"
-                onClick={onDeleteOpen}
-                data-testid="delete-btn"
-              />
-            </DrawerHeader>
-            <DrawerBody>
-              <Text fontSize="sm" mb={4}>
-                {description}
-              </Text>
-              {children}
-            </DrawerBody>
+        <DrawerContent data-testid="edit-drawer-content" py={2}>
+          <Box display="flex" justifyContent="flex-end" mr={2}>
+            <Button variant="ghost" onClick={onClose}>
+              <CloseSolidIcon />
+            </Button>
           </Box>
+          <DrawerHeader py={2} display="flex" alignItems="center">
+            <Text mr="2">{header}</Text>
+            <IconButton
+              aria-label="delete"
+              icon={<TrashCanSolidIcon />}
+              size="xs"
+              onClick={onDeleteOpen}
+              data-testid="delete-btn"
+            />
+          </DrawerHeader>
+          <DrawerBody>
+            <Text fontSize="sm" mb={4}>
+              {description}
+            </Text>
+            {children}
+          </DrawerBody>
+          <DrawerFooter justifyContent="flex-start">
+            <Button onClick={onClose} mr={2} size="sm" variant="outline">
+              Cancel
+            </Button>
+            <Button
+              type="submit"
+              colorScheme="primary"
+              size="sm"
+              data-testid="save-btn"
+              form={formId}
```

Comment:
Yep, form id comes in real handy here, and these changes are an improvement.

I think this also illustrates that EditDrawer is mixing two responsibilities: tidying up the layout markup we don't want to repeat, and managing the close/delete/submit state. It can be helpful to break a component like this into smaller pieces that are focused on the styling so that there are fewer layers to pass props through. Sketch of an example:

```jsx
export const EditCollectionDrawer = () => {
  // ...

  return (
    // maybe hoisting the <form> tag up here doesn't seem as crazy anymore
    <EditDrawer>
      <EditDrawerHeader>{/*...*/}</EditDrawerHeader>

      <EditDrawerBody>{/*...*/}</EditDrawerBody>

      <EditDrawerFooter>
        <CancelButton>
        <SaveButton /* or maybe form={id} could be passed directly ...*//>
      </EditDrawerFooter>
    </EditDrawer>
  );
};
```

Reply:
oh I see what you mean! I'll leave it as-is for now, but will keep that separation in mind for future components (and maybe this one too, if things start getting messier)